tools/CustomMSBuild/Versioning.props

```diff
@@ -18,7 +18,7 @@
     -->
   <PropertyGroup>
     <VersionStartYear Condition="'$(VersionStartYear)' == ''">2020</VersionStartYear>
-    <VersionDateCode>$([System.Convert]::ToInt16('$([MSBuild]::Add(1, $([MSBuild]::Subtract($([System.DateTime]::Now.Year), $(VersionStartYear)))))$([System.DateTime]::Now.ToString("MMdd"))'))</VersionDateCode>
+    <VersionDateCode>$([System.Convert]::ToUInt16('$([MSBuild]::Add(1, $([MSBuild]::Subtract($([System.DateTime]::Now.Year), $(VersionStartYear)))))$([System.DateTime]::Now.ToString("MMdd"))'))</VersionDateCode>

```

Comment:
ToInt16 can't work at your side?

Reply:
@xuzhg It works, but this will last twice as long.

Reply:
I think its better to change the function to "ToUInt16(...) Just like @bdebaere has done or to "ToInt32()". This ensures no future conflict of VersionDateCode value.

Reply:
ToInt16 or ToUInt16 both has the limitation. Sreejith is working on to change the project structure. I'd like to change this logic also in the new project structure.

Reply:
@xuzhg I understand that you want to change the project structure so as to solve this completely. But why not make it a little better in the meantime? I don't see a reason why they should be mutually exclusive, as the possibility exists that this project structure rework will get postponed.

Reply:
@bdebaere @paulodero I'd agree that this change is a better improvement. It's fine to me to merge it.